### DISCUSSION TITLE
Print with speaker notes

### DISCUSSION THREAD

**Discussion by armenic** [google/comprehensive-rust#1041] [Q&A] [answered]
Dear developers, community,

Is there a way to print the book with the speaker notes open? When I click print book icon then the pages are rendered with speakers notes appearing as a collapsed `Details >` and it is not clickable in the PDF. It seems one needs to go page by page expand each speaker note so that they are printed.

Thanks so much for your help!

**Comment by mgeisler** [comment]
Hi @armenic! That's a good question. The state (collapsed, open, popup) is stored in your browser and I would have expected all the notes on the print page to respect this. However, I see now that it's only the first note that is open.

Looking further, I see that I even added [some JS code for the print page](https://github.com/google/comprehensive-rust/blob/main/speaker-notes.js#L129-L138). I'm not sure why it doesn't work as I intended.

Let me create a bug for this.

**Reply by mgeisler** [reply, marked as answer]
@sakex was able to fix it via #1048.

**Reply by armenic** [reply]
You guys are awesome, thank you so much @sakex and @mgeisler. Finally I can read my beloved "Comprehensive Rust" on a subway without keeping dozens of browser tabs open 😆
